magda-web-client/src/Components/Common/DataPreviewMapOpenInNationalMapButton.tsx

```diff
@@ -67,7 +68,52 @@ class DataPreviewMapOpenInNationalMapButton extends Component<PropsType> {
                 ]
             };
         } else {
+            const dataUrl = distribution?.downloadURL
+                ? distribution.downloadURL
+                : distribution.accessURL;
+
+            let layers: any = undefined;
             const id = "data.gov.au-postMessage-" + distribution?.identifier;
+            if (dataUrl) {
+                const queries = URI.parseQuery(dataUrl);
```

Comment:
`URI.parseQuery` is not for the purpose of extracting query string from URL.
It expects a query string input only (i.e. it didn't expect the hostname or other part of the URL string).

I think your logic around here could be simplified as:
```
const queries = typeof dataUrl === "string" ? URI(dataUrl).query(true) : {};
const layers = queries["layers"] ? queries["layers"] : queries["LAYERS"];
```

Please note: URI.query will always return a hash / map for any string input